conjurapi/resource.go

```diff
@@ -16,8 +16,8 @@ type ResourceFilter struct {
 
 // CheckPermission determines whether the authenticated user has a specified privilege
 // on a resource.
-func (c *Client) CheckPermission(resourceID, privilege string) (bool, error) {
-	req, err := c.CheckPermissionRequest(resourceID, privilege)
+func (c *Client) CheckPermission(resourceID string, roleID string, privilege string) (bool, error) {
```

Comment:
Thinking maybe we should retain `CheckPermission` and introduce `CheckPermissionForRole`. Something off about giving special meaning to `roleId` being an empty string.

Reply:
Implemented `CheckPermissionForRole` and `CheckPermissionForRoleRequest`